src/core/api/api_document.pl

```diff
@@ -190,29 +191,43 @@
         Id,
         (   json_read_tail_stream(Tail_Stream, Document),
             nb_thread_var(
-                {Graph_Type, Transaction, Context, Document, Id}/[State, Captures_Out]>>(
-                    api_insert_document_unsafe_(Graph_Type, Transaction, Context, Document, State, Id, Captures_Out)
+                {Graph_Type, JSON, Transaction, Context, Document, Id}/[State, Captures_Out]>>(
+                    api_insert_document_unsafe_(Graph_Type, JSON, Transaction, Context, Document, State, Id, Captures_Out)
                 ),
                 Captures_Var
             )
         ),
         Ids
     ).
-insert_documents_(false, Graph_Type, Stream, Transaction, Captures_Var, Ids) :-
+insert_documents_(false, Graph_Type, JSON, Stream, Transaction, Captures_Var, Ids) :-
     findall(
         Id,
         (   json_read_list_stream(Stream, Document),
             nb_thread_var(
-                {Graph_Type, Transaction, Document, Id}/[State, Captures_Out]>>(
-                    api_insert_document_(Graph_Type, Transaction, Document, State, Id, Captures_Out)
+                {Graph_Type, JSON, Transaction, Document, Id}/[State, Captures_Out]>>(
+                    api_insert_document_(Graph_Type, JSON, Transaction, Document, State, Id, Captures_Out)
                 ),
                 Captures_Var
             )
         ),
         Ids
     ).
 
-api_insert_documents(SystemDB, Auth, Path, Graph_Type, Author, Message, Full_Replace, Stream, Requested_Data_Version, New_Data_Version, Ids) :-
+xor(true,false).
+xor(false,true).
+
+api_insert_documents(SystemDB, Auth, Path, Stream, Requested_Data_Version, New_Data_Version, Ids, Options) :-
+    option(graph_type(Graph_Type), Options),
+    option(author(Author), Options),
+    option(message(Message), Options),
+    option(full_replace(Full_Replace), Options),
+    option(json(JSON), Options),
+    die_if(
+        (   Graph_Type = schema,
+            JSON = true
+        ),
+        error(json_and_schema_disallowed,_)
```

Comment:
This error should be caught, a useful error message should be printed, and a test should be added for that message.